src/Api/Utilities/CommandResultExtensions.cs

```diff
@@ -17,6 +17,18 @@ public static IActionResult MapToActionResult<T>(this CommandResult<T> commandRe
         };
     }
 
+    public static IActionResult MapToActionResultWithSingleErrorMessage<T>(this CommandResult<T> commandResult)
+    {
+        return commandResult switch
+        {
+            NoRecordFoundFailure<T> failure => new ObjectResult(failure.ErrorMessage) { StatusCode = StatusCodes.Status404NotFound },
+            BadRequestFailure<T> failure => new ObjectResult(failure.ErrorMessage) { StatusCode = StatusCodes.Status400BadRequest },
+            Failure<T> failure => new ObjectResult(failure.ErrorMessage) { StatusCode = StatusCodes.Status400BadRequest },
+            Success<T> success => new ObjectResult(success.Value) { StatusCode = StatusCodes.Status200OK },
+            _ => throw new InvalidOperationException($"Unhandled commandResult type: {commandResult.GetType().Name}")
+        };
+    }
+
     public static IActionResult MapToActionResult(this CommandResult commandResult)
```

Comment:
I think we really need to clean up `ErrorMessages`, `Errors` and `ErrorMessage` in `CommandResult`/`Failure`. I think it's causing this unnecessary boilerplate, needing to distinguish between single vs. multiple error messages.

As far as I can tell, `Failure` will only ever contain 1 error message. If we had multiple, it would correspond to a bulk request which would use `Partial`.

Reply:
We discussed this over a call, but I just want to add it here for posterity.

We should remove the array of errors since I don't believe any of our code is currently using it. Another issue is that our client code (at least some) currently only handles a single string for the error message if we pass back an array, it will break the client. I confirmed this with the client that is using this endpoint.